File: 1200. Minimum Absolute Difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        x = float("inf")
        l = []
        z = len(arr)-1
        for i in range(z):
            y = arr[i+1]-arr[i]
            if y<x:
                x = y
        for i in range(z):
            if arr[i+1]-arr[i]==x:
                l.append([arr[i],arr[i+1]])
        return l

File: 1200. Minimum Absolute Difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr);
        int x = Integer.MAX_VALUE;
        int z = arr.length-1;
        List<List<Integer>> lst = new ArrayList<>();
        for(int i=0;i<z;i++){
            int y = arr[i+1]-arr[i];
            if(y<x){
                x = y;
            }
        }
        for(int i=0;i<z;i++){
            if(arr[i+1]-arr[i]==x){
                ArrayList<Integer> pair=new ArrayList<>(2);
                pair.add(arr[i]);
                pair.add(arr[i+1]);
                lst.add(pair);
            }
        }
        return lst;

    }
}
